src/common.jl

```diff
@@ -18,14 +18,14 @@ end
 
 # the result type
 
-immutable Result{T}
+immutable Result{T} <: Real
```

Comment:
I don't think we want `Result` to be a subtype of `Real`. Perhaps you meant `Result{T<:Real}`?